modules/social_features/social_tagging/src/SocialTaggingService.php

```diff
@@ -201,9 +202,22 @@ public function field(
   /**
    * {@inheritdoc}
    */
-  public function groupActive(): bool {
-    return (bool) $this->configFactory->get('social_tagging.settings')
-      ->get('tag_type_group');
+  public function groupActive(Group $group = NULL): bool {
+    // If the group is set we should return TRUE if group bundle is active.
+    if ($group) {
+      return (bool) $this->configFactory->get('social_tagging.settings')
+        ->get('tag_group_type_' . $group->bundle());
+    }
+
+    $configs = $this->configFactory->get('social_tagging.settings')->getRawData();
+    // If the group isn't set we should return TRUE if any group is active.
+    foreach ($configs as $cid => $data) {
+      if (str_contains($cid, 'tag_group_type_')) {
+        return TRUE;
+      }
+    }
+
+    return FALSE;
```

Comment:
Hmm, I think we need to split this up.
You can see from the places where this is used, the idea is

```
  /**
   * Returns whether the feature is turned on for groups or not.
   */
  public function groupActive(): bool;
  ```
  
To check if this feature is enabled for groups (basically any bundle) and that works well for the Override as there we just add stuff to our views/indexes and more. But it doesn't work for places that do care about bundles.

I think we need to think about how this works with the new settings form. Creating some new methods and potentially deprecating this one seems to be better, so that we can better explain how to use it.

Especially because 

```
    [tag_type_group] => 1
    [tag_group_type_flexible_group] => 1
```

the old `tag_type_group` doesn't work anymore and if "any" group bundle is active, is not necessarily the same from a technical type as having it enabled for Groups as a functional object.

Reply:
@ronaldtebrake I've split up this function on two - `groupTypeActive` and `groupsActive`